### ISSUE TITLE
Attack damage resist penetration isn't restricted to attacks for damage calculation

### ISSUE THREAD

**Issue by Demiu** [PathOfBuildingCommunity/PathOfBuilding#3032]
### Check version

- [X] I'm running the latest version of Path of Building and I've verified this by checking the [changelog](https://github.com/PathOfBuildingCommunity/PathOfBuilding/blob/master/CHANGELOG.md)

### Check for duplicates

- [X] I've checked for duplicate issues by using the search function of the [issue tracker](https://github.com/PathOfBuildingCommunity/PathOfBuilding/issues)

### What is the expected behaviour/value?

Devine Fervour notable's damage penetration states "Attack damage penetrates 3% of enemy elemental resistances", so it should only work for attacks.

**Edit:** Commenters on the issue said it's not divine fervour specifically but all "Attack damage penetrates X% elemental resistance" modifiers, so I've updated the title.

### What is the actual behaviour/value?

It's not limited to attacks.

### How to reproduce the issue

1. Pick and add any basic spell to start with like fireball or freezing pulse.
2. Path next to Divine Fervour (next to templar)
3. Hover over it to see the damage increase.
4. Allocate it.
5. Go into calcs tab, change the socket group to that spell, hover over the effective DPS mod category for any element, see that it's increased by 3% by Divine Fervour  

### Build code

```shell
eNrFWltz2kYUfk5_hYZn22h1Vwa3g23sMGMSCnbS9iWzSAuoXiQqrXDIr-_ZlTCyy-WwaaeZjC3Qd-5nz2XHnV--LbixYnmRZOlli1yYLYOlURYn6eyy9fhwex60fvn5p86Qivmn6VWZcPnm55_eddSzwdmKcaBrGYLmMyY-bzjZX4HTkqZizrJ0QP_M8rssvmx9zFLWMiY0jROx-RRxWhQf6YJdth7YYslp3jJoEbE0vt6-qaALmqTjLHpi4i7PyqUSvErY8yCLAdMfDD-NHhpik7QpFrR-1xlyumb5WFBhFPDjstUF4-mMfUgEsKK8BD7Es5wL3wtst9U-SHRDF_DzBLrxkrF4i7_wiGsHTv3z_yca5qw3nbJIJCt2nSfiek7TaGvfXrpTsYOSi2TJE5Y3NNzrtA__YE5Mcx_4IROU3wzHL1hLBsWzXD9Uxv83dNk2efZq9iUR8ysOUdGQImn7szQRTJN4mCVFlv6AfU3SvSZeZ4tJkmpZuCHtrmankg5oSq-zYhsCi-yD3idT9gq615TeGIcbQQHAIaWaQ5ZDXRM4AqnsSQRjsT1Pjr1XY_ZXE7iX2w37tj1wzgFuTeBebv10a4B_SLcmkLjufu1WmYBOg3PjliM5VDGj9xLcT6OGBofYPqY5K1i-alba_QJeU9SRxdQ0SThiM4Y09p6xaH4HnXJEBcPl5QsqJAfdI7FN9xxkusM7-9m_JjjBOZLwtXOg2R0Cn-ieLzSPEeUiZflsPZ4njCPQypdNEpRPmwTIdGiSnGh3b0WL5vki7mFrKjguORi0QSCI2Zu-7v57FMM8-1MOMvw0sm6-yMocGcEKjDJ5OF8XSQSdVE2MIxaXEa56vcxjg2zFFnAm1GAH8-42MPtIrzhMym8MP2QQ5ydRdIWg0dNNFs_YSUJOorhNcvBWkTTa37mHQH-CAf2aLjEjizxMWAFbNFrAfTKbixSWJryUNyR4W-Y0K04wZgtHi7gtOUeNguOn5DVS0Aln7w3zG3FYMHH9vePcMHsGZeZynSwQB6uBHtDtILJ3xLjNWfp9jeb_Co4S0EvjMpfZjZbxluKfYjpttW7Lp_5imeVCfXlNeVQolv10WQojVZtyIT3_NS0XE7leVb-3VaKJXCRF9HVSTqdyfW6BMrna-nu3t73rh_7nXk0yZqpWGVHGOV0WDPbpNOEtI4nrMI-hOkcCgYZNrl6Zj2Oh70Z5Au47DpXrF0JVtREfx8kFFQGDOFDOsC56WC-ZjHiB81JdHo9j1UKJwFX7H8YstWVinM4iukbgXsYOBLbHWTfhssth_DSA9K3aIwYN7SpPJqVAZZTaGRAayNEZYVZzOERlPgZWjyoI8fUkdxxZ9w5M2sGggAzRDZsySGbUQVbF4YE-sbQue5sS11GnqDAKqH13bFFcraG33EqHvrl1eAWArpOXUNliNqUll9__WlKeiHUtc_v9fXWPqb4t5tnzuFwuK07y5Bbg7fv76k2Xi5qJFHPZmlJe1HeLSkd1R9lVI5v6rCxQF5VJGvEyhvWybqK1OE4nUnTLmMkrzeusTEX9hqWyZ8YbK94wljyl2HcdUKTG3vFsQrm14VwZZZmtV6_JhqHqFP1YNTn2HWr_sJS2GH9V9vXVKVceqrFDmgtlyowt5NsBEzSmgrb7AnzRlg5pK93gaTdP2QXf2vXCeuuqqPICPMlOJYftrZJGxbHKpkqcSpUqQ-TjAyANWnlKkZI6PvChebMsTQB1Uuh-INYNHNc-I75vu2fEIS4584hruWeeHXrumW16Iby1TPvMIY5vnVm-6zlnTmi5TssQILFx9Q27UnWrLSW4dZR6MRTW-KMU1q6-eRzdq4d3cyGWxft2-_n5-WJJxTybsm-wuVxE2aK9BDZgyLny0rkU1O7Cv6vuY7f7afR59BjZo69pHJ8_9gqv36Xe9-4fs8tLJaC9kdCpLs2LSlz9SRkuNSSeRQLIT1FFtZ5TduAIcU0EziOB7WBwjhvYCJxlhSGGnxN4fojhZxOCgIWOifGK7Xkoa23HwVlLwgDDz3JCjBW27bsYsTbxcOp5vuUicK5LQkyyOKEZYHDEJj4G54ceSqwXBBYKB3ag3OzZmHA4pmOaqHCYGLHE8oiNMsMOUdGFbPEw0Q3dAJMFxAlRckM_9DHqBY6PEWuFPrFQyUxCD1XRbNNF4awgQFU030HVFjsMQpRcz8PYQXyL4Cokugb5uI7goOQ6kM-YdHEJFEmMvaZrY_KABI6NSntQ0EcVNS_AdSzfQ9lhEQ9VXQgwxOAc4qHyyrJR4TVhaEK1yhBVJL2QoGqk5e_KFhgFt9MOfICpTw2IciyUDx8zWALlO_nl5oMaGj8n7NkoGM2j-Vjkcib_nmWL3-Xlt31hmqbjWaH8X20CHxgVA7qsh22JrLcIp94hYN-_SWBBzdX-sxl3JfA3ydK58EPTsS0zDHy_2nfUHF2PrvJ5zKopuCwY7E1ZGn9hdJml6msptbrSqYB7QWq12uwKY55trn8qmEHGz3RpdCfroqDcqF28I5q7SN-SWTiyw6irLBOFpkK7bDmi1BXj4kSSW5jtnwwbpZK1SyVbz704J1j_Epmjp6SnJ83T96ZOPh2JwAfGF0ycKmhHJnk6co6kH6zxK3mvfDpjW_8UER15jl4uEL3Mc46HB-PYU9nuSi9Nw1F1x9XQR6MG2sfExGujuoM8MXy7vayVX64OEepIYgKhZaer7VZLI-6e3kmy9dLXPj3LHP2yr5HTRNv5NiYhHAxIQ-0fGH6QkdSPgqeRlfph-IF0wZZkSYs8NXiohudG8q7b0ikxlrZ7NcsFPp465tgIPxH9MDja3nJ1rMGcF_0kP6JSd1HyHb2t0653SLUnqyVU_X1Blk6TGSA67bd___836RaCBg==
```

### Screenshots

Since I'm having problem with github's image upload, [an imgur link to illustrate](https://i.imgur.com/RDReGJJ.png).

**Comment by Switchersaw** [comment]
Following on to this, all sources of 'Attack Damage Penetrates xx Elemental Resistances' are not being restricted to attacks. Forces of Nature is the same, and I made a custom item with the same wording and it was also working for spells and not restricted to attacks.

I tested Spell Damage Penetrates xx Resistances and found no issue with it being restricted to spells.

**Comment by jais6278** [comment]
Similar issue involving "Primeval Force" node, but all "Attack damage penetrates #% of enemy elemental resistances" are behaving this way

**Comment by Lothrik** [comment]
Fixed by #3310.
